salt/kubernetes-minion/stop.sls

```diff
@@ -0,0 +1,31 @@
+# Stop and disable the Kubernetes minion daemons, ensuring pods have been drained
+# and the node is marked as unschedulable.
+
+# If this, or the state below, fails we should ignore it and proceed anyway as
+# Kubernetes will recover.
+cordon-kubelet:
+  cmd.run:
+    - name: |
+        kubectl --kubeconfig /var/lib/kubelet/kubeconfig cordon {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }}
+    - check_cmd:
+      - /bin/true
+
+drain-kubelet:
+  cmd.run:
+    - name: |
+        kubectl --kubeconfig /var/lib/kubelet/kubeconfig drain {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }} --ignore-daemonsets --grace-period=300 --timeout=340s
```

Comment:
I've always thought `drain` implies a `cordon`. From the [documentation for `drain`](https://kubernetes.io/docs/user-guide/kubectl/v1.6/#drain): _"Drain node in preparation for maintenance.The given node will be marked unschedulable to prevent new pods from arriving."_